### PR TITLE
Refactored/Reformatted heart lung monitor, parking lot and other actor codebase

### PR DESCRIPTION
**Pull request by mavericksid**

### CODE REVIEW

File: src/main/scala/com/knoldus/akka/actor/ParkingLot.scala [hunk 1]
```diff
@@ -1,52 +1,58 @@
...
+
 class SlotMonitor extends Actor {
   def receive = {
     case "GiveMeEmptySlot" => sender ! 5
-    case x => println("Slot Monitor")
+    case msg               => println("Slot Monitor")
   }
 }
 
+object Attendant {
```
Comment (satendrakumar) [inline src/main/scala/com/knoldus/akka/actor/ParkingLot.scala]:
```Scala
object Attendant{
  def props(slotMonitor :ActorRef) = Props(classOf[Attendant], slotMonitor)
}

````

File: src/main/scala/com/knoldus/akka/actor/ParkingLot.scala [hunk 2]
```diff
@@ -1,52 +1,58 @@
...
 
   def receive = {
-    case "LetMePark" => {
+    case "LetMePark" =>
       println("Got the message to park")
-      val parkingSlot = Await.result((Init.slotMonitor ? "GiveMeEmptySlot").mapTo[Int], 5.seconds)
-      println(s"parking at ${parkingSlot}")
-    }
-    case x => println("Attendant")
+      val parkingSlot = Await.result((Init.slotMonitor ? "GiveMeEmptySlot").mapTo[Int], timeout.duration)
```
Comment (satendrakumar) [inline src/main/scala/com/knoldus/akka/actor/ParkingLot.scala]: This is no right way to do this. 
problems
 1) Cyclic dependency 
 2)  tight coupling(Not testable)
 3) Blocking
Solution: 
```scala
class Attendant(slotMonitor :ActorRef) extends Actor {
  import context.dispatcher

  implicit val timeout = Timeout(5.seconds)

  def receive = {
    case "LetMePark" => {
      println("Got the message to park")
      val parkingSlotFuture = (slotMonitor ? "GiveMeEmptySlot").mapTo[Int]
      parkingSlotFuture.onComplete{
        case Success(parkingSlot) =>
         println(s"parking at ${parkingSlot}")
        case Failure(th) =>
          th.printStackTrace()
      }
      //more elegant =>  parkingSlotFuture pipeTo sender()
    }
    case x => println("Attendant")
  }
}
```

File: src/main/scala/com/knoldus/akka/actor/ParkingLot.scala [hunk 3]
```diff
@@ -1,52 +1,58 @@
...
-
 object Init {
   val system = ActorSystem("PL")
-  val driver = system.actorOf(Props[Driver], "Driver")
-  val slotMonitor = system.actorOf(Props[SlotMonitor], "SlotMonitor")
-  val attendant = system.actorOf(Props[Attendant], "Attendant")
+
+  val driver = system.actorOf(Driver.props(), "Driver")
+  val attendant = system.actorOf(Attendant.props(), "Attendant")
+  val slotMonitor = system.actorOf(SlotMonitor.props(), "SlotMonitor")
```
Comment (satendrakumar) [inline src/main/scala/com/knoldus/akka/actor/ParkingLot.scala]:
```Scala
   val slotMonitor = system.actorOf(SlotMonitor.props(), "SlotMonitor")
   val attendant = system.actorOf(Attendant.props(slotMonitor), "Attendant")
```

File: src/main/scala/com/knoldus/akka/actor/ParkingLot.scala (hunk 2 above)
Comment (mavericksid) [inline src/main/scala/com/knoldus/akka/actor/ParkingLot.scala]: Hi @satendrakumar, nice suggestions!
It'd be helpful if you can include them as a part of a separate or this pull request.

Comment (satendrakumar) [inline src/main/scala/com/knoldus/akka/actor/ParkingLot.scala]: @mavericksid Sure

**Pull request merged**